### ISSUE TITLE
Unable to Sign Typed Data V4 or Sign Permit

### ISSUE THREAD

**Issue by 0xGitGuy** [trezor/trezor-suite#9365]
**Describe the bug**
Metamask throws "Result: Error: Unknown message" when attempting to Sign Typed Data V4 or Sign Permit. 

Trezor Protected Wallet address:
Sign Typed Data V4:  "Result: Error: Unknown message"
Sign Permit : "Result: Error: Unknown message"
Personal Sign: No issues

NON-Trezor Wallet address:
Sign Typed Data V4:  No issues
Sign Permit : No issues
Personal Sign: No issues

**Firmware version and revision**
Model One: 1.11.1 and 1.12.1
Trezor Suite : v23.8.1 

**Desktop/smartphone setup (please complete the following information):**
 - Metamask 10.35.0 and 10.35.1
 - Windows 10
 - Brave Version 1.57.57 Chromium: 116.0.5845.163 (Official Build) (64-bit)

**To Reproduce**
1. Simply perform any Typed Data V4 or Sign Permit
2. https://metamask.github.io/test-dapp/

**Expected behavior**
Data signing should not produce error

**Comment by matejcik** [comment]
hello Connect pals! please check why this case calls `EthereumSignTypedData` (not supported on T1) instead of `EthereumSignTypedHash`. it is possible that metamask is not providing the needed data, but Connect should not try to call EthereumSignTypedData (instead return an error like "data for t1 is missing") when the device is a T1 anyway, no?

**Comment by 0xGitGuy** [comment]
It's a Model One. Browser dev console shows:
`{code: -32603, message: 'Unknown message', data: {…}}`

Seems to suggest it's an Internal JSON-RPC error as per: 
https://eips.ethereum.org/EIPS/eip-1474#error-codes

How do we go from here?

**Comment by mroz22** [comment]
it got broken by this, trying to fix this asap https://github.com/trezor/trezor-suite/pull/9104/files#diff-ec4c42097058d399dae4d75a562e5f8f33449cbd52742da9b67904618aa145f7R119

cc @matejkriz

**Comment by mroz22** [comment]
please check now, hotfix should be online.

**Comment by 0xGitGuy** [comment]
First time updating custom firmware. Does it wipe Model One's memory just like when downgrading the firmware? Otherwise, normally when's the next official firmware update that can be done seamless on Trezor Suite?

**Comment by Hannsek** [comment]
Yes, it does wipe the memory.

**Comment by Hannsek** [comment]
For T1, it is not planned anytime soon.

**Comment by matejcik** [comment]
more to the point, is your fw upgrade relevant to this issue in any way?

**Comment by 0xGitGuy** [comment]
The hotfix is supposed to solve the error  "Result: Error: Unknown message" upon performing Sign Permit, am I right? If not, i'm missing something here.

Typically what do i lose if the device memory is wiped? I'm weighing the trade offs vs not being able to use my favorite dapps. A few of them have moved to using Permit and I prefer that compared to granting Approve()

**Comment by 0xGitGuy** [comment]
Okay, more info to help with this case here:

Oddly, I used an older MetaMask version 10.30.4, disabled the most recent MetaMask version. 10.30.4 managed to sign Permits and Typed Data v4 without errors.

After this, I disabled MetaMask 10.30.4, and re-enabled the latest MetaMask 10.35.1. Signing Permits and Typed Data v4 magically worked. Trezor device firmware remained the same throughout, no change.

No idea what went wrong here or if the firmware hotfix would actually fix anything since I did not apply it.
